pkg/pillar/types/types.go

```diff
@@ -280,62 +328,121 @@ type UEvent struct {
 	Env    map[string]string
 }
 
-// UUIDPairAndIfIdxToNum used for appNum on network instance
+// AppInterfaceKey uniquely references application interface.
+type AppInterfaceKey struct {
+	// NetInstID : UUID of the network instance to which the interface is connected.
+	// Previously it was called BaseID, which was rather unspecific and a bit confusing.
+	// In order to not break EVE upgrades, the json tag is set to use this old field name.
+	NetInstID uuid.UUID `json:"BaseID"`
+	// AppID : UUID of the application
+	AppID uuid.UUID
+	// IfIdx : interface index among interfaces of this app on this network instance.
+	// First has index 0, next 1, etc.
+	IfIdx uint32
+}
+
+// Key returns string representation of the key.
+func (info AppInterfaceKey) Key() string {
+	return fmt.Sprintf("%s-%s-%d",
+		info.NetInstID.String(), info.AppID.String(), info.IfIdx)
+}
+
+// AppInterfaceToNum is used to store number assigned to an application interface.
+// The number is unique only within the network instance where the interface
+// is connected to. It is used to generate an IP address for the interface.
+type AppInterfaceToNum = UUIDPairAndIfIdxToNum
+
+// UUIDPairAndIfIdxToNum is a legacy name for a structure holding number allocated
+// to an application interface. Because this name is rather unspecific and quite
+// clumsy, we now use the AppInterfaceToNum alias instead.
+// However, in order to not break EVE upgrades (by changing pubsub topic name),
+// we continue using this name for the underlying structure.
```

Comment:
Does this mean that the topic in the /persist/status will continue to be called UUIDPairAndIfIdxToNum??

Reply:
Yes, correct, type has not changed, only how we as developers refer to it.

Reply:
Can you add some text to specifically say that this ensures that the type=topic in /persist/status remains unchanged as UUIDPairAndIfIdxToNum?